internal/pkg/docker/orchestrator/orchestrator.go

```diff
@@ -185,6 +222,178 @@ func (a *runTaskAction) Do(o *Orchestrator) error {
 	return nil
 }
 
+func (o *Orchestrator) setupProxyConnections(ctx context.Context, pauseContainer string, a *runTaskAction) error {
+	fmt.Printf("\nSetting up proxy connections...\n")
+	g, gctx := errgroup.WithContext(ctx)
+	ports := make(map[Host]string)
+	portsMu := &sync.Mutex{}
+
+	for _, host := range a.hosts {
+		host := host
```

Comment:
https://go.dev/doc/effective_go#sharing

> It may seem odd to write
>
> ```
> req := req
> ```
> but it's legal and idiomatic in Go to do this. You get a fresh version of the variable with the same name, deliberately shadowing the loop variable locally but unique to each goroutine.